Spigot-API-Patches/0210-Support-components-in-ItemMeta.patch

```diff
@@ -0,0 +1,77 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MiniDigger <admin@minidigger.me>
+Date: Sat, 6 Jun 2020 18:13:16 +0200
+Subject: [PATCH] Support components in ItemMeta
+
+
+diff --git a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
+index 7e0b1d363829031075b6822d7b35ccde9fc05da9..5db838cab09888c3b3537e122e0e381a78e1dbb7 100644
+--- a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
++++ b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
+@@ -42,6 +42,18 @@ public interface ItemMeta extends Cloneable, ConfigurationSerializable, Persiste
+     @NotNull
+     String getDisplayName();
+ 
++    // Paper start
++    /**
++     * Gets the display name that is set.
++     * <p>
++     * Plugins should check that hasDisplayName() returns <code>true</code>
++     * before calling this method.
++     *
++     * @return the display name that is set
++     */
++    @NotNull
++    net.md_5.bungee.api.chat.BaseComponent[] getDisplayNameComponent();
```

Comment:
I'd personally prefer none of these methods have `Component` in the names. The param type difference is enough. ;)

Reply:
tell that the java compiler, lol.
cant have two methods with same params (void) called getDisplayName.
Am open for suggestions, but adding the component suffix was the next best thing.

Reply:
Perhaps it would be more natural to call it 'getDisplayedName' so as to not conflict with getDisplayName

Reply:
> cant have two methods with same params (void) called getDisplayName

Good call :3

I meant more for the setters. But I guess then it just doesnt match the getters.

Reply:
Returntype overloading isn't supported by the Java language. The JVM supported or probably supports it until today. But to keep it clean I do not recommend doing this hacky things. Remember the getOnlinePlayers "hack". There was a time where the server supported plugin using getOnlinePlayer(): Player[] and getOnlinePlayers(): Collection<? extends Player>